modules/platforms/dotnet/Apache.Ignite.Core/Impl/PlatformJniTarget.cs

```diff
@@ -309,7 +309,8 @@ private void FinishMarshal(BinaryWriter writer)
 
             var fut = convertFunc == null && futType != FutureType.Object
                 ? new Future<T>()
-                : new Future<T>(new FutureConverter<T>(_marsh, keepBinary, convertFunc));
+                : new Future<T>(new FutureConverter<T>(_marsh, keepBinary, _marsh.IsRegisterSameJavaType(),
```

Comment:
We pass `_marsh` to the constructor, no need to pass a property of `_marsh` separately (here and below)

Reply:
Fixed